client/src/Utils/sigmfMetadata.ts

```diff
@@ -0,0 +1,271 @@
+  }
+  getFileName() {
+    return this.global['traceability:origin'].file_path.split('/').pop();
+  }
+  getEmail() {
+    const emailName = this.getAuthor();
+    let author = emailName.split('<');
+    let email: any;
+    if (author.length === 1) {
+      email = null;
+    } else {
+      email = author[1].split('>');
+      email = email[0].trim();
+    }
+    return email;
+  }
+  getFullFilePath() {
+    const origin = this.global['traceability:origin'];
+    const type = origin?.type ?? 'local';
+    if (type === 'local') {
+      return origin.file_path;
+    } else if (type === 'azure_blob') {
+      return `https://${origin.account}.blob.core.windows.net/${origin.container}/${origin.file_path}`;
+    } else {
+      return `${origin.account}/${origin.container}/${origin.file_path}`;
+    }
+  }
+
+  getCanonicalFilePath() {
+    const origin = this.global['traceability:origin'];
+    if (origin) {
+      return `${origin.account}/${origin.container}/${origin.file_path}`;
+    }
+    return '';
+  }
+
+  getShortDescription() {
+    let shortDescription = this.global['core:description'] ?? '';
+    shortDescription = shortDescription.substring(0, 60);
+    if (shortDescription.length === 60) {
+      shortDescription += '...';
+    }
+    return shortDescription;
+  }
+
+  getThumbnailUrl() {
+    return this.getFullFilePath() + '.jpeg';
+  }
+  getDataUrl() {
+    return this.getFullFilePath() + '.sigmf-data';
+  }
+  getMetadataUrl() {
+    return this.getFullFilePath() + '.sigmf-meta';
+  }
+  getDataType() {
+    return this.global['core:datatype'];
+  }
+  getDataTypeDescription() {
+    return this.global['core:datatype']
+      .replace('c', 'complex\n')
+      .replace('r', 'real\n')
+      .replace('f', 'float\n')
+      .replace('i', 'signed int\n')
+      .replace('u', 'unsigned int\n')
+      .replace('8', '8 bits')
+      .replace('16', '16 bits')
+      .replace('32', '32 bits')
+      .replace('64', '64 bits')
+      .replace('_le', '');
+  }
+  getDescription() {
+    return this.global['core:description'] ?? '';
+  }
+
+  getLengthInMillionIQSamples() {
+    return (this.global['traceability:sample_length'] ?? 0) / 1e6;
+  }
+
+  getLengthInIQSamples() {
+    return this.global['traceability:sample_length'] ?? 0;
+  }
+
+  getLengthInBytes() {
+    return this.getLengthInIQSamples() * this.getBytesPerSample();
+  }
+
+  getCenterFrequency() {
+    for (let i = 0; i < this.captures.length; i++) {
+      if (this.captures[i]['core:frequency']) {
+        return this.captures[i]['core:frequency'];
+      }
+    }
+    return 0;
+  }
+}
+
+export class CaptureSegment {
+  'core:sample_start': number;
+  'core:global_index'?: number;
+  'core:header_bytes'?: number;
+  'core:frequency'?: number;
+  'core:datetime'?: string;
+  [key: string]: any;
+}
+
+export class TraceabilityOrigin {
+  type: string;
+  account: string;
+  container: string;
+  file_path: string;
+}
+
+export class Annotation {
```

Comment:
Is it worth exporting this into it's own file? this file's a bit of a chunky boi

Reply:
Yeha I thought about that the file kept growing but changin it now would require to change the imports of the changes that were done already in the other commit so I would say this is something that could be done after